includes/classes/JSONScrubber.php

```diff
@@ -0,0 +1,378 @@
+				$scrub = $this->scrub_object_by_type( $revision_id, $post_type, 'revision' );
+
+				if ( $this->show_errors && ( false === $scrub || is_wp_error( $scrub ) ) ) {
+					WP_CLI::error( "Unable to scrub revision ID: {$revision_id}" );
+				}
+
+				$progress->tick();
+			}
+
+			$progress->finish();
+		}
+	}
+
+	/**
+	 * Scrub taxonomies.
+	 */
+	public function scrub_taxonomies() {
+		if ( empty( $this->config->taxonomies ) ) {
+			return;
+		}
+
+		foreach ( $this->config->taxonomies as $taxonomy ) {
+			$term_ids = Helpers\get_all_term_ids_of_taxonomy( $taxonomy->name );
+			$progress = \WP_CLI\Utils\make_progress_bar( "Scrubbing {$taxonomy->name} terms", count( $term_ids ) );
+
+			foreach ( $term_ids as $term_id ) {
+				$scrub = $this->scrub_object_by_type( $term_id, $taxonomy, 'term' );
+
+				if ( $this->show_errors && ( false === $scrub || is_wp_error( $scrub ) ) ) {
+					WP_CLI::error( "Unable to scrub term ID: {$term_id}" );
+				}
+
+				$progress->tick();
+			}
+
+			$progress->finish();
+		}
+	}
+
+	/**
+	 * Scrub options.
+	 */
+	public function scrub_options() {
+		global $wpdb;
+
+		if ( empty( $this->config->options ) ) {
+			return;
+		}
+
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Scrubbing options', count( $this->config->options ) );
+
+		foreach ( $this->config->options as $option ) {
+
+			if ( 'remove' === $option->action ) {
+				$wpdb->delete( $wpdb->options, [ 'option_name' => $option->name ] );
+
+			} else {
+				$wpdb->update(
+					$wpdb->options,
+					[ 'option_value' => $this->get_field_data_by_action( $option ) ],
+					[ 'option_name' => $option->name ]
+				);
+			}
+
+			$progress->tick();
+		}
+
+		$progress->finish();
+	}
+
+	/**
+	 * Scrub custom tables.
+	 */
+	public function scrub_custom_tables() {
+		global $wpdb;
+
+		if ( empty( $this->config->custom_tables ) ) {
+			return;
+		}
+
+		foreach ( $this->config->custom_tables as $table ) {
+			$name     = $table->name;
+			$pk       = $table->primary_key;
+			$query    = "SELECT {$pk} FROM {$name}";
+			$ids      = $wpdb->get_col( $query ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+			$progress = \WP_CLI\Utils\make_progress_bar( "Scrubbing {$name} table", count( $ids ) );
+
+			foreach ( $ids as $id ) {
+				$new_data = [];
+
+				foreach ( $table->columns as $field ) {
+					$new_data[ $field->name ] = $this->get_field_data_by_action( $field );
+				}
+
+				$wpdb->update( $name, $new_data, [ $pk => $id ] );
+
+				$progress->tick();
+			}
+
+			$progress->finish();
+		}
+	}
+
+	/**
+	 * Truncate tables.
+	 */
+	public function truncate_tables() {
+		global $wpdb;
+
+		if ( empty( $this->config->truncate_tables ) ) {
+			return;
+		}
+
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Truncating tables', count( $this->config->truncate_tables ) );
+
+		foreach ( $this->config->truncate_tables as $table ) {
+			$wpdb->query( "TRUNCATE TABLE {$table}" ); // phpcs:ignore WordPress.DB.PreparedSQL
+			$progress->tick();
+		}
```

Comment:
**issue:** If we add `wp_comments` as a truncate table, it doesn't auto-translate to `wp_2_comments` on multisite setups. We should look into this.

Reply:
In general, we should look at multi-site support. Since this is reasonably common within our setups and we _should_ be able to check whether it's a MS based upon the config.

Reply:
I wonder if we could do something like the below. Then within the `truncate_tables` option, we could use `wp_*_comments` to tell it to strip all comment tables 🤔 

```suggestion
		foreach ( $this->config->truncate_tables as $table ) {
			$table = str_replace( 'wp_*_', $wpdb->prefix, $table );
			$wpdb->query( "TRUNCATE TABLE {$table}" ); // phpcs:ignore WordPress.DB.PreparedSQL
			$progress->tick();
		}
```

Alternatively, we could always swap out `wp_` for `$wpdb->prefix` 🤔